### PR TITLE
added Comment status

### LINKED ISSUE #5153: [Change request to #3120] <Edit comment>
**Reason for Change:**
To clarify some System behavior & text field "Edited"

**Mockup**
https://tinyurl.com/2s3nrm6v

**Acceptance Criteria**
**6.**  When "Save changes" is tapped then the System displays the already edited comment with a text field "Edited" before a comment date
comment_edited

**7.** If an User during the comment editing click on some URL on the comments field, the user can continue editing his comment

**Impact of Change**
#3120

Epic: #3194

### PR DESCRIPTION
**Pull request by ospodaryk**
# GreenCity PR
[[Change request to #3120] <Edit comment> #5153](https://github.com/ita-social-projects/GreenCity/issues/5153)

## Summary Of Changes :fire:
Added eventComment status and modified date
Added econewsComment status 

## Added
eventComment status
eventComment  modified date
econewsComment status

## Changed
methods where we change anything with eventComment

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: dao/src/main/java/greencity/entity/event/EventComment.java [hunk 1]
```diff
@@ -22,6 +23,9 @@
 import javax.persistence.JoinTable;
 import javax.persistence.Transient;
 import javax.persistence.CascadeType;
+import javax.persistence.Enumerated;
+import javax.persistence.EnumType;
+
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/entity/event/EventComment.java]: needless blank line

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 2]
```diff
@@ -227,7 +229,7 @@ public void delete(Long eventCommentId, UserVO user) {
         if (eventComment.getComments() != null) {
             eventComment.getComments().forEach(comment -> comment.setDeleted(true));
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java]: should we set `EventCommentStatus.DELETED` for every comment here too?

**Comment by RomaMocherniuk** [comment]
please add tests for the new logic

File: dao/src/main/java/greencity/entity/event/EventComment.java (hunk 1 above)
Comment (ospodaryk) [reply dao/src/main/java/greencity/entity/event/EventComment.java]: fixed

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java (hunk 2 above)
Comment (ospodaryk) [reply service/src/main/java/greencity/service/EventCommentServiceImpl.java]: fixed

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 3]
```diff
@@ -222,12 +224,15 @@ public void delete(Long eventCommentId, UserVO user) {
...
         }
-
         eventComment.setDeleted(true);
+        eventComment.setStatus(EventCommentStatus.DELETED);
         if (eventComment.getComments() != null) {
-            eventComment.getComments().forEach(comment -> comment.setDeleted(true));
+            eventComment.getComments()
+                .forEach(comment -> {
+                    comment.setDeleted(true);
+                    comment.setStatus(EventCommentStatus.DELETED);
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java]: please add test for this case too

**Comment by RomaMocherniuk** [comment]
please fix code smells and increase test coverage
image

**Review by RomaMocherniuk — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Review by 2or5 — APPROVED** [review]

**Review by HelenSotnik — APPROVED** [review]

**Pull request merged**